app/repositories/VatSchemeRepository.scala

```diff
@@ -236,9 +240,51 @@ class VatSchemeRepository @Inject() (
       .toFuture()
       .map(_ => status)
 
-  private[repositories] def registrationSelector(regId: String, internalId: Option[String] = None) =
+  private[repositories] def registrationSelector(regId: String, internalId: Option[String] = None) = {
+
+    val baseFilter: Bson = equal("registrationId", regId)
+
+    val regFilter: Bson = internalId match {
+      case Some(id) =>
+        and(
+          baseFilter,
+          equal(internalIdKey, id),
+          or(
+            equal("status", "submitted"),
+            and (
+              equal("status", "draft"),
+              gt("timestamp", cutoffDate)
+            )
+          )
+        )
+      case None =>
+        baseFilter
+    }
+    regFilter
+  }
+
+  private[repositories] def registrationSelectorNewApp(regId: String, internalId: Option[String] = None) = {
     internalId
       .map(intId => and(equal("registrationId", regId), equal("internalId", intId)))
       .getOrElse(equal("registrationId", regId))
+  }
+
+  // Extract the document for a specific user(internalId) provided if its in submitted status or its still a draft but satisfies the cutOffDate
+  private def regAllSelector(internalId: String) = {
```

Comment:
This should probably be something like 'internalIdAndDateFilterSelector' rather than 'regAllSelector'.
Maybe we also separate the filters out of the selectors to make them more obvious, e.g.:
`def internalIdFilter(internalId: String): Bson = equal(internalIdKey, internalId)`
`def registrationIdFilter(registrationId: String): Bson = equal(registrationIdKey, registrationId)`
`def draftDateFilter: Bson = or (equal ("status", "submitted"), and(equal ("status", "draft"), gt ("timestamp", cutoffDate)))`
And the selectors can then call these filters